tests/phpunit/tests/db.php

```diff
@@ -1717,6 +1729,36 @@ public function data_prepare_with_placeholders() {
 				false,
 				"'hello' 'foo##'",
 			),
+			array(
+				'SELECT * FROM %i WHERE %i = %d;',
+				array( 'my_table', 'my_field', 321 ),
+				false,
+				'SELECT * FROM `my_table` WHERE `my_field` = 321;',
+			),
+			array(
+				'WHERE %i = %d;',
+				array( 'evil_`_field', 321 ),
```

Comment:
I'd prefer to see more tests, even if they seem obvious and redundant, like multiple escaping. They may not seem relevant now but might catch regressions in future:

`evil_````````_field`
` ``evil_field`` `
`evil'field`
`evil_\``_field`
`evil_%s_field`

etc

Reply:
Thank you, and I agree, I'm just not sure how many tests each project likes to have. I've added your suggested tests. The last one, which uses placeholder_escape is especially useful. I've tried to think of some more, but nothing springs to mind.

Reply:
I'll try to think of some more and suggest additions. Unbalanced backtick quotes maybe?

Reply:
Good idea, I've added those as both unbalanced in the `$query` string, and one in the value (which is just an extra check to prove that it's working really) - [commit](https://github.com/WordPress/wordpress-develop/pull/2192/commits/488bd374f40a831aea529f30ca95a3524f120c03).